### PR TITLE
Do not mark tests xfailed if cluster doesn't come up in time

### PR DESCRIPTION
**Pull request by fjetter**
Not the most elegant solution but should do the trick

### CODE REVIEW

File: distributed/utils_test.py [hunk 1]
```diff
@@ -491,14 +495,20 @@ def run_worker(q, scheduler_q, config, **kwargs):
...
+                    else:
+                        q.put(worker.address)
+                        await worker.finished()
+
+                # Scheduler might've failed
+                if isinstance(scheduler_addr, str):
+                    try:
+                        loop.run_sync(_)
+                    finally:
+                        loop.close(all_fds=True)
```
Comment (crusaderky) [inline distributed/utils_test.py:511]:
```suggestion
                        loop.close(all_fds=True)
                else:
                    assert isinstance(scheduler_addr, Exception)
```

File: distributed/utils_test.py [hunk 2]
```diff
@@ -508,14 +518,20 @@ def run_nanny(q, scheduler_q, config, **kwargs):
...
```
Comment (crusaderky) [inline distributed/utils_test.py:534]:
```suggestion
                        loop.close(all_fds=True)
                else:
                    assert isinstance(scheduler_addr, Exception)
```

File: distributed/utils_test.py (hunk 1 above)
Comment (fjetter) [reply distributed/utils_test.py:511]: We would never receive this exception.

**Comment by fjetter** [comment]
The timed out job actually finished running the test suite as expected without problems, all green. https://github.com/dask/distributed/runs/5248080147?check_suite_focus=true

However, is hang during teardown

```
= 1344 passed, 59 skipped, 1201 deselected, 9 xfailed, 2 xpassed, 17 warnings, 80 leaked in 1714.67s (0:28:34) =
Traceback (most recent call last):
  File "/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/asyncio/selector_events.py", line 918, in write
    n = self._sock.send(data)
BrokenPipeError: [Errno 32] Broken pipe

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/weakref.py", line 667, in _exitfunc
    f()
  File "/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/weakref.py", line 591, in __call__
    return info.func(*info.args, **(info.kwargs or {}))
  File "/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/site-packages/asyncssh/process.py", line 1342, in kill
    self._chan.kill()
  File "/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/site-packages/asyncssh/channel.py", line 1433, in kill
    self.send_signal('KILL')
  File "/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/site-packages/asyncssh/channel.py", line 1399, in send_signal
    self._send_request(b'signal', String(signal))
  File "/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/site-packages/asyncssh/channel.py", line 716, in _send_request
    self.send_packet(MSG_CHANNEL_REQUEST, String(request),
  File "/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/site-packages/asyncssh/channel.py", line 710, in send_packet
    self._conn.send_packet(pkttype, payload, handler=self)
  File "/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/site-packages/asyncssh/connection.py", line 1572, in send_packet
    self._send(packet + mac)
  File "/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/site-packages/asyncssh/connection.py", line 1349, in _send
    self._transport.write(data)
  File "/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/asyncio/selector_events.py", line 924, in write
    self._fatal_error(exc, 'Fatal write error on socket transport')
  File "/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/asyncio/selector_events.py", line 719, in _fatal_error
    self._force_close(exc)
  File "/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/asyncio/selector_events.py", line 731, in _force_close
    self._loop.call_soon(self._call_connection_lost, exc)
  File "/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/asyncio/base_events.py", line 746, in call_soon
    self._check_closed()
  File "/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/asyncio/base_events.py", line 510, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
Error: The operation was canceled.
```

Anybody seen this before?

**Comment by fjetter** [comment]
This stuck test job appears to be an old problem, see https://github.com/dask/distributed/pull/2925
I'm not sure why we're hitting this now again

see https://github.com/ronf/asyncssh/issues/112

https://github.com/dask/distributed/pull/3885 for an attempt to implement the hotfix

**Review by crusaderky — APPROVED** [review]

**Comment by fjetter** [comment]
I'll move on with this change since I believe raising the exceptions explicitly is valuable and I don't see how the ssh problem could connect to the changes I'm proposing but if it happens again after merging we can revert this again.

**Pull request merged**